### DISCUSSION TITLE
Getting 100+ errors when including single header

### DISCUSSION THREAD

**Discussion by Pedroma34** [skypjack/entt#725] [Q&A] [answered]
When I include "entt.hpp" I get 100+ errors. error E0457  basic_registry

**Comment by skypjack** [comment, marked as answer]
My wild guess is that you're not using at least C++17. However, it's hard to tell since you haven't provided enough context to help you out.

**Reply by Pedroma34** [reply]
> My wild guess is that you're not using at least C++17. However, it's hard to tell since you haven't provided enough context to help you out.

Sorry, I was frustrated so I didn't provide enough detail. I'm currently using Visual Studio 2019. I downloaded the solo header file and included into my project. I get around 124 error messages. I'm clearly not including it the way it's supposed to be included. I'm may be missing the "...Then pass the proper -I argument to the compiler to add the src directory to the include paths..." part. How do I do that in visual studio?

**Reply by skypjack** [reply]
Did you set C++17 as the language revision to use for your project? Otherwise, these errors are quite expected.

**Reply by Pedroma34** [reply]
> Did you set C++17 as the language revision to use for your project? Otherwise, these errors are quite expected.

Thank you that worked. Also, I had to disable warning 4996 for a deprecated function. Thank you.
